googlesearch/__init__.py

```diff
@@ -83,9 +87,7 @@
 # If the cookie jar is inaccessible, the errors are ignored.
 home_folder = os.getenv('HOME')
 if not home_folder:
-    home_folder = os.getenv('USERHOME')
-    if not home_folder:
-        home_folder = '.'   # Use the current folder on error.
+    home_folder = os.getenv('USERHOME') if home_folder else '.' # Use the current folder on error.
```

Comment:
Have you tested this? Looks to me like it won't work, since the if statement would be evaluated before the assignment

Reply:
this refactoring changes the logic and it's not correct.

inside we enter the conditional in the case the env var HOME is null or an empty string (or the integer zero, or an empty iterable).
so first we obtain the env var USERNAME, then we check if USERHOME is null or something. in this case we set home_folder as current folder.


in the refactored logic, this changes a bit:
we enter the conditional when the env var HOME is null.
then here is the mistake:
we check again if the home_folder is null, and if it's null we assign it to the value of the env var USERHOME, otherwise (if it's not null) we assign it to the current folder (.).
this is incorrect because:
```
home_folder = os.getenv('USERHOME') if home_folder else '.'
```
is equivalent to:
```
if home_folder:
    home_folder = os.getenv('USERHOME')
else:
    home_folder = '.'
```
So:
1) the first thing we do inside the conditional is to evaluate conditional's expression again under the same conditions.
2) logic is broken, what we want for **home_folder** is that if HOME is null, then try to assign USERHOME and if it's also null, fall to '.'.
2.1) if HOME is null, then the result after the conditional is the USERHOME value, which can be null and in this case we want it to be dot.
2.2) if HOME is not null, we will never enter this conditional, so the else is useless.

Reply:
if you still would like to use a ternary operator to write this expression, the correct code is:
```
home_folder = os.getenv('USERHOME') if os.getenv('USERHOME') else '.'
```

but i don't see how this would make the code more clean, as it makes the code more difficult to read and understand.